python/ray/data/context.py

```diff
@@ -37,6 +38,14 @@
     os.environ.get("RAY_DATASET_PUSH_BASED_SHUFFLE", None)
 )
 
+# Whether to use Polars for tabular dataset groupbys and aggregations.
+try:
+    import polars as pl  # noqa
```

Comment:
I think we should make this explicit opt-in to avoid strange performance differences based on what packages happened to be installed.